### PR TITLE
Remove session checks to allow for a token authenticated user to be set.

### PR DESCRIPTION
**Pull request by bmarika**

### CODE REVIEW

**Comment by bmarika** [comment]
Resolves https://github.com/etianen/django-reversion/issues/524

File: src/reversion/middleware.py [hunk 1]
```diff
@@ -29,9 +29,7 @@ def _close_revision(self, request):
 
     def process_response(self, request, response):
         """Closes the revision."""
-        # look to see if the session has been accessed before looking for user to stop Vary: Cookie
-        if hasattr(request, 'session') and request.session.accessed \
-                and hasattr(request, "user") and request.user is not None and request.user.is_authenticated() \
+        if hasattr(request, "user") and request.user is not None and request.user.is_authenticated() \
```
Comment (etianen) [inline src/reversion/middleware.py]: Are we absolutely certain that this won't result in the session being accessed?

I was actually wondering about making the check simply inspect request.method, and set the user if it's a POST, PATCH or PUT request. It's simpler, and mutating methods shouldn't suffer from losing the cache.

**Comment by bmarika** [comment]
@etianen Am also not sure on whether the session will be accessed.  Let me make the change and check for the request.method

**Comment by etianen** [comment]
Thanks! Just merged!

**Pull request merged**